src/Rings/mpoly-localizations.jl

```diff
@@ -1699,152 +1734,210 @@ function Base.show(io::IO, I::MPolyLocalizedIdeal)
   print(io, last(gens(I)))
 end
 
+@attr function shifted_ideal(
+    I::MPolyLocalizedIdeal{LRT, LRET}
+  ) where {LRT<:MPolyLocalizedRing{<:Any, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, LRET}
+  L = base_ring(I)
+  R = base_ring(L)
+  shift, tfihs = base_ring_shifts(L)
+  return ideal(R, shift.(gens(pre_saturated_ideal(I))))
+end
+
+function Base.in(
+    a::RingElem, 
+    I::MPolyLocalizedIdeal{LocRingType}
+  ) where {
+    LocRingType<:MPolyLocalizedRing{<:Any, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}
+  }
+  L = base_ring(I)
+  parent(a) == L || return L(a) in I
+  L = base_ring(I)
+  R = base_ring(L)
+  shift, tfihs = base_ring_shifts(L)
+  Is = shifted_ideal(I)
+  # We have to call for that groebner basis once manually. 
+  # Otherwise the ideal membership will complain about the ordering not being global.
+  o = negdegrevlex(gens(R))
+  groebner_basis(Is, ordering=o, enforce_global_ordering=false)
+  return ideal_membership(shift(numerator(a)), Is, ordering=o)
+end
+
+function coordinates(
+    a::RingElem,
+    I::MPolyLocalizedIdeal{LRT} 
+  ) where {LRT<:MPolyLocalizedRing{<:Any, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}}
+  L = base_ring(I)
+  L == parent(a) || return coordinates(L(a), I)
+  R = base_ring(L)
+  J = shifted_ideal(I)
+  shift, tfihs = base_ring_shifts(L)
+  p = shift(numerator(a))
+  o = negdegrevlex(gens(R))
+  x, u = Oscar.lift(p, J, o)
+  T = pre_saturation_data(I)
+  return L(one(base_ring(L)), tfihs(u)*denominator(a), check=false)*change_base_ring(L, map_entries(tfihs,x))*T
+end
+
 ########################################################################
-# Groebner and standard bases                                          #
-########################################################################
+# The next method is based on the following observation: 
 #
-# A "groebner basis" for an ideal I in a localized ring W = R[S⁻¹] in 
-# is a LocalizedBiPolyArray G for which the following holds.
+#     p//q ∈ I ⋅ U⁻¹ ⋅ V⁻¹ ⊂ R[U⁻¹⋅V⁻¹] 
+#   ⇔ ∃ u ∈ U, v∈ V : u ⋅ v ⋅ p ∈ I
+#   ⇔ ∃ v ∈ V : v ⋅ p ∈ I : U 
 #
-# G has attached to it an ordering and a singular ring. The reduction 
-# of the numerator a of an element a//b of a localized ring W = R[S⁻¹] 
-# by the elements of G with respect to that ordering has to be zero 
-# if and only if a//b belongs to the ideal. 
-
-### The catchall implementation. 
-# This refers back to the saturated ideal procedures and ideal 
-# membership of the numerator in the base_ring.
-@Markdown.doc """
-    groebner_basis(
-        I::MPolyLocalizedIdeal;
-        ordering::MonomialOrdering = default_ordering(I)
-      )
-
-For an ideal ``I ⊂ R[S⁻¹]`` in a localized polynomial ring generated 
-by fractions ``a₁//b₁,…, aₘ//bₘ``, this returns a groebner basis 
-for the ideal ``I' ⊂ R`` generated by the numerators ``a₁
-"""
-function groebner_basis(
```

Comment:
Sorry, for being two weeks late, but why are the `groebner_basis` functions gone? Will they come back?

Reply:
The `groebner_basis` function should not be gone! At least not for polynomial ideals. But for ideals in localizations, there simply is no groebner basis in general. Introducing them was once motivated by the fact that for the particular case of the localization at the origin, you may speak of standard basis. But in general, we can not hope for any such thing. 

Instead, we followed the general philosophy according to the Posur [paper](https://arxiv.org/abs/1709.08180) on localizations that we have already applied to the modules. The groebner basis command for localized ideals will not come back.

For localizations at orderings, the philosophy is now the following: There is a `base_ring(W)` for a localized ring `W` and a `I = pre_saturated_ideal(J)` for an ideal `J` in `W`. Now you can call `groebner_basis(I, ordering=my_ordering, enforce_global_ordering=false)` from @ederc 's package. I hope to provide a small pr soon which gives a wrapper for this so that you can simply call `standard_basis(I, my_ordering)` instead.

Reply:
Ah, that sounds reasonable. Thank you for the explanation!